src/ripple/protocol/impl/tokens.cpp

```diff
@@ -241,5 +261,358 @@ decodeBase58Token(std::string const& s, TokenType type)
     // Skip the leading type byte and the trailing checksum.
     return ret.substr(1, ret.size() - 1 - guard.size());
 }
+}  // namespace b58_ref
+
+#ifndef _MSC_VER
+namespace b58_fast {
+namespace detail {
+B58Result<std::span<std::uint8_t>>
+b256_to_b58(std::span<std::uint8_t const> input, std::span<std::uint8_t> out)
+{
+    // Allocate enough base 58^10 coeff for encoding 38 bytes
+    // (33 bytes for nodepublic + 1 byte token + 4 bytes checksum)
+    // log(2^(38*8),58^10)) ~= 5.18. So 6 coeff are enough
+    if (input.size() > 38)
+    {
+        return Unexpected(TokenCodecErrc::InputTooLarge);
+    };
+
+    auto count_leading_zeros = [&](auto const& col) -> std::size_t {
+        std::size_t count = 0;
+        for (auto const& c : col)
+        {
+            if (c != 0)
+            {
+                return count;
+            }
+            count += 1;
+        }
+        return count;
+    };
+
+    auto const input_zeros = count_leading_zeros(input);
+    input = input.subspan(input_zeros);
+
+    std::array<std::uint64_t, 6> base_58_10_coeff{};
+    std::array<std::uint64_t, 5> base_2_64_coeff_buf{};
+    std::span<std::uint64_t> const base_2_64_coeff =
+        [&]() -> std::span<std::uint64_t> {
+        // convert from input from big endian to native u64, lowest coeff first
+        std::size_t num_coeff = 0;
+        for (int i = 0; i < 5; ++i)
+        {
+            if (i * 8 > input.size())
+            {
+                break;
+            }
+            auto const src_i_end = input.size() - i * 8;
+            if (src_i_end >= 8)
+            {
+                std::memcpy(
+                    &base_2_64_coeff_buf[num_coeff], &input[src_i_end - 8], 8);
+                boost::endian::big_to_native_inplace(
+                    base_2_64_coeff_buf[num_coeff]);
+            }
+            else
+            {
+                std::uint64_t be = 0;
+                for (int bi = 0; bi < src_i_end; ++bi)
+                {
+                    be <<= 8;
+                    be |= input[bi];
+                }
+                base_2_64_coeff_buf[num_coeff] = be;
+            };
+            num_coeff += 1;
+        }
+        return std::span(base_2_64_coeff_buf.data(), num_coeff);
+    }();
+
+    constexpr std::uint64_t B_58_10 = 430804206899405824;  // 58^10;
```

Comment:
I don't think we need the `ull`. I doubt we'll ever port rippled to a system where we need the `ull` (and in the very unlikely case we do it will quickly break unit tests).